src/swap.ts

```diff
@@ -5,6 +5,11 @@ import { Field } from "./cnd/siren";
 import { sleep, timeoutPromise, TryParams } from "./util/timeout_promise";
 import { AllWallets, Wallets } from "./wallet";
 
+/**
+ * A helper type to return errors in case of failure with a wallet or blockchain action.
+ */
+export type Result = string | Error;
```

Comment:
Imho this is too rusty for JS. I think it's more common to just throw or return a tuple i.e. `(returnValue, error)` where both can be undefined. 

I'm happy to go with your proposal though if you feel this is the right way.

Reply:
> Imho this is too rusty for JS. I think it's more common to just throw or return a tuple i.e. `(returnValue, error)` where either is defined.

Your tuple is just a different manner to define a data type `Result`.